lib/optimizely/config_manager/http_project_config_manager.rb

```diff
@@ -178,9 +183,14 @@ def request_config
         return
       end
 
-      @last_modified = response[Helpers::Constants::HTTP_HEADERS['LAST_MODIFIED']]
-
-      config = DatafileProjectConfig.create(response.body, @logger, @error_handler, @skip_json_validation) if response.body
+      if response_code >= 200 && response_code < 400
+        @logger.log(Logger::DEBUG, 'Successfully fetched datafile, generating Project config')
+        config = DatafileProjectConfig.create(response.body, @logger, @error_handler, @skip_json_validation)
+        @last_modified = response[Helpers::Constants::HTTP_HEADERS['LAST_MODIFIED']]
```

Comment:
what happens if last modified is nil?

Reply:
It is assigned as nil. I just moved this line of code inside the check. assuming it was working correctly earlier as well without a `nil` check